### PR TITLE
Show local and remote exceptions during preview.

### PR DESCRIPTION
**Pull request by scheglov**
image

### CODE REVIEW

File: src/io/flutter/preview/PreviewView.java [hunk 1]
```diff
@@ -177,14 +180,10 @@ public void onResponse(FlutterOutline widget, JsonObject response) {
     }
 
     @Override
-    public void onFailure(RenderProblemKind kind, @Nullable FlutterOutline widget) {
+    public void onFailure(RenderProblemKind kind, FlutterOutline widget) {
```
Comment (devoncarew) [inline src/io/flutter/preview/PreviewView.java]: should widget now be marked `@Nonnull`?

File: src/io/flutter/preview/PreviewView.java [hunk 2]
```diff
@@ -900,6 +915,46 @@ private void showNotRenderableInPreviewArea(@NotNull FlutterOutline widget) {
     setSplitterProportion(0.9f);
   }
 
+  private void showLocalException(@NotNull Throwable localException) {
+    final JPanel panel = new JPanel();
+    panel.setLayout(new MigLayout("insets 0", "[grow, center]", "[grow, bottom][grow 200, top]"));
+
+    panel.add(new JBLabel("Encountered an exception during rendering"), "cell 0 0");
+
+    final LinkLabel linkLabel = LinkLabel.create("Show the exception...", () -> {
```
Comment (devoncarew) [inline src/io/flutter/preview/PreviewView.java]: perhaps just `Show exception...`?

File: src/io/flutter/preview/PreviewView.java [hunk 3]
```diff
@@ -900,6 +915,46 @@ private void showNotRenderableInPreviewArea(@NotNull FlutterOutline widget) {
...
+      final StringWriter stringWriter = new StringWriter();
+      final PrintWriter printWriter = new PrintWriter(stringWriter);
+      printWriter.println(localException.getMessage());
+      localException.printStackTrace(printWriter);
+      printWriter.println();
+      printWriter.println();
```
Comment (devoncarew) [inline src/io/flutter/preview/PreviewView.java]: having two println()s are intentional?

File: src/io/flutter/preview/PreviewView.java (hunk 1 above)
Comment (scheglov) [reply src/io/flutter/preview/PreviewView.java]: No, it is still `@Nullable`.
I restored the annotation.

**Pull request merged**